examples/concurrent_executions/concurrent_executions.c

```diff
@@ -0,0 +1,193 @@
+/*
+  This is free and unencumbered software released into the public domain.
+
+  Anyone is free to copy, modify, publish, use, compile, sell, or
+  distribute this software, either in source code form or as a compiled
+  binary, for any purpose, commercial or non-commercial, and by any
+  means.
+
+  In jurisdictions that recognize copyright laws, the author or authors
+  of this software dedicate any and all copyright interest in the
+  software to the public domain. We make this dedication for the benefit
+  of the public at large and to the detriment of our heirs and
+  successors. We intend this dedication to be an overt act of
+  relinquishment in perpetuity of all present and future rights to this
+  software under copyright law.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+  IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+
+  For more information, please refer to <http://unlicense.org/>
+*/
+
+#include <stdio.h>
+#include <string.h>
+
+#include "cassandra.h"
+
+#include <uv.h>
+
+#define CONCURRENCY_LEVEL 32 /* Maximum amount of parallel async executions (threads) */
```

Comment:
This should probably be `NUM_THREADS` and should be much smaller than 32 or we could just eliminate threads entirely from this example to simplify it to focus on doing concurrent asynchronous requests. I think that `CONCURRENCY_LEVEL` is a better name for outstanding asynchronous requests.

Reply:
Maybe a 3rd way is using synchronous requests from multiple threads.

Reply:
I chose the chunking method because I wanted to keep things off the callback for our examples as a "best practice" approach.